resolwe_bio/processes/reads_processing/bbduk.py

```diff
@@ -0,0 +1,1275 @@
+                default=50,
+                description="To use the sliding window set minimum entropy in range between 0.0 and 1.0.",
+            )
+            entropy_k = IntegerField(
+                label="Length of kmers used to calcuate entropy",
+                default=5,
+            )
+            entropy_mask = BooleanField(
+                label="Mask low-entropy parts of sequences with N instead of discarding",
+                default=False,
+            )
+            min_base_frequency = IntegerField(
+                label="Minimum base frequency",
+                default=0,
+            )
+
+        class Fastqc:
+            """FastQC parameters."""
+
+            nogroup = BooleanField(
+                label="Disable grouping of bases for reads >50bp",
+                default=False,
+                description="All reports will show data for every base in the read. Using this option "
+                "will cause fastqc to crash and burn if you use it on really long reads.",
+            )
+
+        reference = GroupField(Reference, label="Reference", hidden="!show_advanced")
+
+        processing = GroupField(
+            Processing, label="Processing parameters", hidden="!show_advanced"
+        )
+
+        operations = GroupField(
+            Operations,
+            label="Trimming, filtering and masking parameters.",
+            hidden="!show_advanced",
+        )
+
+        header_parsing = GroupField(
+            HeaderParsing, label="Header-parsing parameters", hidden="!show_advanced"
+        )
+
+        complexity = GroupField(
+            Complexity, label="Complexity parameters", hidden="!show_advanced"
+        )
+
+        fastqc = GroupField(Fastqc, label="FastQC parameters", hidden="!show_advanced")
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label="Remaining upstream reads")
+        fastq2 = ListField(FileField(), label="Remaining downstream reads")
+        statistics = ListField(FileField(), label="Statistics")
+        fastqc_url = ListField(
+            FileHtmlField(), label="Upstream quality control with FastQC"
+        )
+        fastqc_url2 = ListField(
+            FileHtmlField(), label="Downstream quality control with FastQC"
+        )
+        fastqc_archive = ListField(
+            FileField(), label="Download upstream FastQC archive"
+        )
+        fastqc_archive2 = ListField(
+            FileField(), label="Download downstream FastQC archive"
+        )
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+        reads_fw_basename = Path(inputs.reads.output.fastq[0].path).name
+        assert reads_fw_basename.endswith(".fastq.gz")
+        name_fw = reads_fw_basename[:-9]
+        reads_rv_basename = Path(inputs.reads.output.fastq2[0].path).name
+        assert reads_rv_basename.endswith(".fastq.gz")
+        name_rv = reads_rv_basename[:-9]
+
+        input_fw_reads = "input_fw_reads.fastq.gz"
```

Comment:
The naming of the reads and variables was not optimal in the original process. Please use mate1/mate2 terminology instead, and replace the relevant lines, for example `input_fw_reads = "input_fw_reads.fastq.gz"` could be  `input_mate1_reads = "input_mate1_reads.fastq.gz"`